transcom/mymove
# Main B-19805

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-19805)

## INT PR
#12718 

## Summary

This branch adds the frontend code that complements #12669. Now when reviewing documents during PPM Closeout, the SC can edit the "Expense Type" and "Description" fields. A dropdown box has been added for the type, and a text field for description. These pre-populate with the info we receive from the backend, so that the SC will not need to touch these fields if no changes are needed. 

### How to test

1. Create a PPM move and go through the normal process to get it to PPM closeout. On the customer side, be sure to create some expenses (third/last section in the receipt upload page) for the SC to review.
2. Login as an SC and access the move. Click Review Documents.
3. Accept the Trip ticket and any pro gear tickets. Once you get to Expense receipts, you should now see the new editable fields. Make sure that they have pre-populated with the correct values (i.e. the ones the customer wrote in during the upload of PPM Documents).
4. If you edit either of these, the changes are only saved if you click "Continue". An easy way to check if they are uploaded to the DB is to edit, click "Continue", then immediately click "Back". The fields should now pre-populate to the values that you submitted instead of the original.
5. If you want to be extra sure that they have updated, you can find the expense in the `moving_expense` table in DBeaver. You can find their ID in the network tab, or can just search by description if you put something unique in.
6. Check that values are not saved if you click "Back".
7. The document count specific to each type (I.E. "Contracted Expense #&#8203;1") should also update as well. E.g. If you have two tickets, one "Oil" and one "Tolls", each will say "{expenseType} #&#8203;1". However, if you change the "Oil" ticket to "Tolls", when clicking "Continue" you should now see the next "Tolls" receipt say "Tolls #&#8203;2" at the bottom of the review card.

## Screenshots
![Screenshot 2024-05-17 at 5 18 41 PM](https://github.com/transcom/mymove/assets/146959895/2de1e6d4-ff12-4c87-b1da-70d2942c9ed3)
![Screenshot 2024-05-17 at 5 18 49 PM](https://github.com/transcom/mymove/assets/146959895/b38bf342-38b2-40a4-a597-80fabbcd734e)
![Screenshot 2024-05-17 at 5 18 58 PM](https://github.com/transcom/mymove/assets/146959895/bcaa79ce-bc17-4ed7-b726-f43c127bf9ea)
